### PR TITLE
Add a new doc to introduce how to customize monitor servers in tiup environment

### PR DESCRIPTION
**Pull request by qqqdan**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [X] master (the latest development version)
- [X] v5.4 (TiDB 5.4 versions)
- [X] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by srstack — APPROVED** [review]

**Review by nexustar — APPROVED** [review]

File: TOC.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 - 关于 TiDB
   - [TiDB 简介](/overview.md)
-  - [TiDB 5.4 Release Notes](/releases/release-5.4.0.md)
+  - [TiDB 5.3 Release Notes](/releases/release-5.3.0.md)
```
Comment (nexustar) [inline TOC.md:6]: I guess it is unesserary here

**Comment by TomShawn** [comment]
从文档角度，建议只为最新的一两个版本提供新文档~

File: tiup/customized-montior-in-tiup-environment.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+上述配置后，在集群进行 deploy/scale-out/scale-in/reload 操作时， TiUP 会将 additional_scrape_conf 字段的内容会添加到 Prometheus 配置文件的对应参数中。
+
+# 自定义 Grafana 配置
+
+目前 TiUP 可以支持自定义 Grafana Dashboard 和其他配置。
+
+## 1. 自定义 Grafana Dashboard
+
+TiUP 支持自定义 Grafana Dashboard ，需要将自定义的 Dashboard 配置文件放到 TiUP 所在机器的某个目录下， 并在 TiDB 集群的拓扑配置 topology.yaml 文件中设置自定义规则文件目录 dashboard_dir 为放置 Dashboard 配置文件的目录。
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:85]:
```suggestion
1. 将自定义的 Dashboard 配置文件放到 TiUP 所在机器的某个目录下。

2. 在 TiDB 集群的拓扑配置 topology.yaml 文件中，将自定义规则文件目录 dashboard_dir 设置为实际
放置 Dashboard 配置文件的目录。
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+...
+# # Server configs are used to specify the configuration of Grafana Servers.
+grafana_servers:
+  # # The ip address of the Grafana Server.
+  - host: 127.0.0.1
+    dashboard_dir: /home/tidb/dashboards   # grafana dashboard dir on TiUP machine
+...
+```
+
+上述配置后，在集群进行 deploy/scale-out/scale-in/reload 操作时， TiUP 将读取**本机** /home/tidb/dashboards 路径下的自定义 Dashboard ，然后将该配置发送到 Grafana Server， 替换默认配置规则。
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:99]:
```suggestion
3. 在集群进行 deploy/scale-out/scale-in/reload 操作，TiUP 读取**本机** /home/tidb/dashboards 路径下的自定义 Dashboard，然后将该配置发送到 Grafana Server， 替换默认的配置规则。
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+## 2. 自定义 Grafana 其他配置
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:101]:
```suggestion
### 自定义 Grafana 其他配置
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+当需要自定义 Grafana 的其他配置，不能直接修改 Grafana server 的配置文件（/usr/local/etc/grafana/grafana.ini）该文件会在集群进行 deploy/scale-out/scale-in/reload 操作时被 TiUP 重置。 需要将相关自定义配置写入集群配置文件 topology.yaml 中的 grafana_servers 的配置部分。
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:103]:
```suggestion
1. 打开集群配置文件 topology.yaml。

2.  在 grafana_servers 的配置部分添加其他配置。
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+  - host: 127.0.0.1
+    config:
+      log.file.level: warning
+      smtp.enabled: true
+      smtp.host: {IP}:{port}
+      smtp.user: example@pingcap.com
+      smtp.password: {password}
+      smtp.skip_verify: true
+...
+```
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:121]: 代码格式需要修改。

File: tiup/customized-montior-in-tiup-environment.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+上述配置后，在集群进行 deploy/scale-out/scale-in/reload 操作时， TiUP 会将 config 字段的内容会添加到 grafana 的配置文件 grafana.ini 中。
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:123]:
```suggestion
3. 在集群进行 deploy/scale-out/scale-in/reload 操作，TiUP 将 config 字段的内容添加到 grafana 的配置文件 grafana.ini 中。
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+# 自定义 Alertmanager 配置
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:125]:
```suggestion
## 自定义 Alertmanager 配置
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+
+目前 TiUP 可以支持自定义 alertmanager 的监听地址配置。
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:127]:
```suggestion
目前，TiUP 支持自定义配置 Alertmanager 的监听地址。
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+
+TiUP 部署的 Alertmanager 默认监听 alertmanager_servers.host，如果客户的 Alertmanager 是通过代理访问的，则无法访问。通过在集群配置文件 topology.yaml 中添加 listen_host 指定监听地址，使得 Alertmanager 可以通过代理访问。 推荐配置为 0.0.0.0。
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:129]:
```suggestion
TiUP 部署的 Alertmanager 默认监听 alertmanager_servers.host，如果你使用代理，则无法访问 Alertmanager。此时，你可以在集群配置文件 topology.yaml 中添加 listen_host 指定监听地址，使得 Alertmanager 可以通过代理访问。 推荐配置为 0.0.0.0。
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+
+以下例子配置了 listen_host 字段。
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:131]:
```suggestion
以下示例将 listen_host 字段设置为 `0.0.0.0`。
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+
+alertmanager_servers:
+  # # The ip address of the Alertmanager Server.
+  - host: 172.16.7.147
+    listen_host: 0.0.0.0
+    # # SSH port of the server.
+    ssh_port: 22
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:138]: 代码格式需要修改。

File: tiup/customized-montior-in-tiup-environment.md [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+
+上述配置后，在集群进行 deploy/scale-out/scale-in/reload 操作时， TiUP 会将 listen_host 字段的内容会添加到 Alertmanager 启动参数的 '--web.listen-address' 中。
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:140]:
```suggestion
3. 在集群进行 deploy/scale-out/scale-in/reload 操作，TiUP 将 listen_host 字段的内容添加到 Alertmanager 启动参数的 '--web.listen-address' 中。
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+...
+# # Server configs are used to specify the configuration of Prometheus Server.
+monitoring_servers:
+  # # The ip address of the Monitoring Server.
+  - host: 127.0.0.1
+    rule_dir: /home/tidb/prometheus_rule   # prometheus rule dir on TiUP machine
+...
+```
+
+上述配置后，在集群进行 deploy/scale-out/scale-in/reload 操作时， TiUP 将读取**本机** /home/tidb/prometheus_rule 路径下的自定义 rule，然后将该配置发送到 Prometheus Server， 替换默认配置规则。
```
[the remark this answers is not in the document]
Comment (qqqdan) [reply tiup/customized-montior-in-tiup-environment.md:44]: 这个不是用户操作，这个是修改配置后的结果

File: tiup/customized-montior-in-tiup-environment.md [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+
+目前，TiUP 支持自定义 Grafana Dashboard 和其他配置。
+
+### 自定义 Grafana Dashboard
+
+1. 将自定义的 Dashboard 配置文件放到 TiUP 所在机器的某个目录下。
+
+2. 在 TiDB 集群的拓扑配置 topology.yaml 文件中，将自定义规则文件目录 dashboard_dir 设置为实际放置 Dashboard 配置文件的目录。
+
+topology.yaml 文件中 monitoring_servers 的配置示例：
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:87]:
```suggestion
以下为 topology.yaml 文件中的 monitoring_servers 配置示例：
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+
+上述配置后，在集群进行 deploy/scale-out/scale-in/reload 操作时， TiUP 将读取**本机** /home/tidb/dashboards 路径下的自定义 Dashboard ，然后将该配置发送到 Grafana Server， 替换默认配置规则。
+
+### 自定义 Grafana 其他配置
+
+1. 打开集群配置文件 topology.yaml。
+
+2. 在 grafana_servers 的配置部分添加其他配置。
+
+以下例子配置了 [log.file] level 字段以及 smtp 的相关配置项。
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:105]:
```suggestion
以下为 topology.yaml 文件中的 [log.file] level 字段以及 smtp 配置示例：
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+
+上述配置后，在集群进行 deploy/scale-out/scale-in/reload 操作时， TiUP 将读取**本机** /home/tidb/prometheus_rule 路径下的自定义 rule，然后将该配置发送到 Prometheus Server， 替换默认配置规则。
+
+### 自定义 Prometheus scrape 配置
+
+1. 打开 TiDB 集群的拓扑配置文件 topology.yaml。
+
+2. 在 monitoring_servers 的配置部分添加 additional_scrape_conf 字段。
+
+topology.yaml 文件中 monitoring_servers 的配置示例:
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:50]:
```suggestion
以下为 topology.yaml 文件中的 monitoring_servers 配置示例:
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+
+目前，TiUP 支持自定义 Prometheus 的 rule 配置和 scrape 配置。
+
+### 自定义 Prometheus rule
+
+1. 将自定义的 rule 配置文件放到 TiUP 所在机器的某个目录下。
+
+2. 在 TiDB 集群的拓扑配置 topology.yaml 文件中，将自定义规则文件目录 rule_dir 设置为实际 rule 配置文件的目录。
+
+topology.yaml 文件中 monitoring_servers 的配置示例：
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:32]:
```suggestion
以下为 topology.yaml 文件中的 monitoring_servers 配置示例:
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 19]
```diff
@@ -0,0 +1,138 @@
...
+summary: 了解如何自定义 TiUP 管理的监控组件的配置。
+---
+
+# 监控组件的自定义配置
+
+使用 TiUP 部署 TiDB 集群时，TiUP 会同时自动部署监控组件，并且在集群扩容中自动为新增节点添加监控配置。
+
+需要注意的是，TiUP 会使用自己的配置参数覆盖监控组件的配置，如果你直接修改监控组件的配置文件，修改的配置文件可能在对集群进行 deploy/scale-out/scale-in/reload 等操作中被 TiUP 所覆盖，导致配置不生效。
+
+如果需要自定义监控组件的配置，可以参考本文在 TiDB 集群的拓扑配置 topology.yaml 文件中添加对应的配置项。
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:12]:
```suggestion
使用 TiUP 部署 TiDB 集群时，TiUP 会同时自动部署 Prometheus、Grafana 和 Alertmanager 等监控组件，并且在集群扩容中自动为新增节点添加监控配置。

需要注意的是，TiUP 会使用自己的配置参数覆盖监控组件的配置，如果你直接修改监控组件的配置文件，修改的配置文件可能在对集群进行 deploy/scale-out/scale-in/reload 等操作中被 TiUP 所覆盖，导致配置不生效。

如果需要自定义 Prometheus、Grafana 和 Alertmanager 等监控组件的配置，可以参考本文在 TiDB 集群的拓扑配置 topology.yaml 文件中添加对应的配置项。
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: TOC.md [hunk 20]
```diff
@@ -84,6 +84,7 @@
   - [将 Grafana 监控数据导出成快照](/exporting-grafana-snapshots.md)
   - [TiDB 集群报警规则与处理方法](/alert-rules.md)
   - [TiFlash 报警规则与处理方法](/tiflash/tiflash-alert-rules.md)
+  - [监控组件的自定义配置](/tiup/customized-montior-in-tiup-environment.md)
```
Comment (shichun-0415) [inline TOC.md:87]:
```suggestion
  - [自定义监控组件的配置](/tiup/customized-montior-in-tiup-environment.md)
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 21]
```diff
@@ -0,0 +1,138 @@
+---
+title: 监控组件的自定义配置
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:2]:
```suggestion
title: 自定义监控组件的配置
```

File: tiup/customized-montior-in-tiup-environment.md [hunk 22]
```diff
@@ -0,0 +1,138 @@
...
+summary: 了解如何自定义 TiUP 管理的监控组件的配置。
+---
+
+# 监控组件的自定义配置
```
Comment (shichun-0415) [inline tiup/customized-montior-in-tiup-environment.md:6]:
```suggestion
# 自定义监控组件的配置
```

**Pull request merged**